### DISCUSSION TITLE
How to make the side reactions (SEI, Li plating, etc) in the interface dependent of stoichiometry

### DISCUSSION THREAD

**Discussion by Brushnie33** [pybamm-team/PyBaMM#3893] [Q&A]
It is straightforward to change a parameter from pybamm.Parameter to pybamm.FunctionParameter, but the challenge is to specify the sto parameter in the side reactions in the interface folder, for instance, if I would like to make the "SEI kinetic rate constant [m.s-1]" to be dependent of sto, I could define it in the class of ParticleLithiumIonParameters(BaseParameters):

    def k_sei(self, sto):

        domain, Domain = self.domain_Domain
        inputs = {
            f"{Domain} particle stoichiometry": sto,
        }
        return pybamm.FunctionParameter(
            f"{self.phase_prefactor}SEI kinetic rate constant [m.s-1]",
            inputs,
        )

And in the sei_growth.py codes, if I add the following codes:
sto = variables[f"{Domain} particle concentration"] 
And change phase_param.k_sei to phase_param.k_sei(sto) in the following equation:
k_exp = phase_param.k_sei(sto) * pybamm.exp(-alpha_SEI * F_RT * eta_SEI)

And in the parameter file, I added the function:
![image](https://github.com/pybamm-team/PyBaMM/assets/112187928/2d4cedbf-1966-4062-b67b-4d381ee1daa3)

And change the value for the k_sei:
"SEI kinetic rate constant [m.s-1]": k_sei_function, # 1e-12,

I got errors:
![image](https://github.com/pybamm-team/PyBaMM/assets/112187928/14afe704-92fd-4146-8aff-044b9f45bc01)

**Comment by fsenoner** [comment]
Hi, I got a similar problem recently. Did you find a solution for the domain children problem?

**Reply by Brushnie33** [reply]
Yes, in the get_coupled_variables(self, variables) function in the sei_growth.py, I added the following lines:
sto = variables[f"{Domain} particle concentration"] # added for sto dependency
k_sei_soc=  pybamm.r_average(phase_param.k_sei_soc(sto))
k_exp = k_sei_soc* pybamm.exp(-alpha_SEI * F_RT * eta_SEI)

and commented out the original k_exp calculation: # k_exp = phase_param.k_sei* pybamm.exp(-alpha_SEI * F_RT * eta_SEI)

And in the lithium_ion_parameters.py file, I added the definition of k_sei_soc in the ParticleLithiumIonParameters(BaseParameters):
   def k_sei_soc(self, sto):
        Domain = self.domain.capitalize()
        tol = pybamm.settings.tolerances["U__c_s"]
        sto = pybamm.maximum(pybamm.minimum(sto, 1 - tol), tol)
        inputs = {f"{self.phase_prefactor}{Domain} particle stoichiometry": sto}

        return pybamm.FunctionParameter(
            f"{self.phase_prefactor}{Domain} SEI kinetic rate constant user defined [m.s-1]",
            inputs,
        )
